roughpy_jax/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+project(RoughPyJaxModule LANGUAGES CXX)
+
+# Optionally enable CUDA build
+include(CheckLanguage)
+check_language(CUDA)
+
+if (ROUGHPY_JAX_CUDA)
+    if (NOT DEFINED CMAKE_CUDA_COMPILER)
+        message(FATAL_ERROR "CMAKE_CUDA_COMPILER must be available for ROUGHPY_JAX_CUDA")
+    endif()
+
+    enable_language(CUDA)
+    set(CMAKE_CUDA_STANDARD 17)
+    set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+    set(CMAKE_CUDA_SEPARABLE_COMPILATION ON)
+    if (NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+        set(CMAKE_CUDA_ARCHITECTURES 80 86 90 120)
+    endif()
+
+    find_package(CUDAToolkit REQUIRED)
+endif()
+
+# Get XLA includes via jax lib
+find_package(Python 3.8 REQUIRED COMPONENTS Interpreter Development.Module)
+execute_process(
+    COMMAND "${Python_EXECUTABLE}"
+            "-c" "from jax import ffi; print(ffi.include_dir())"
+    OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE XLA_DIR
+)
+message(STATUS "XLA include directory: ${XLA_DIR}")
+
+# FIXME for review whether to use nanobind or native python bindings
+find_package(nanobind CONFIG REQUIRED)
+
+nanobind_add_module(RoughPyJaxModule NB_STATIC 
+    src/cpu/dense_ft_fma.cpp
+    src/py_module.cpp
+)
+
+# Ensure that XLA FFI definitions are visible in genreated .so 
+set_target_properties(RoughPyJaxModule PROPERTIES
+    CXX_VISIBILITY_PRESET default
+)
+
+target_include_directories(RoughPyJaxModule PRIVATE
+    ${XLA_DIR}
+    ${CMAKE_CURRENT_LIST_DIR}/src
+    ${CMAKE_CURRENT_LIST_DIR}/..
+)
+
+set_target_properties(RoughPyJaxModule PROPERTIES
```

Comment:
We need to add install directives here to make sure the roughpy_jax code is copied into the wheel. Check the roughpy CMakeLists for example

Reply:
Thanks @inakleinbottle. I see there are several commented-out lines of code in the original (i.e. WIN32 and APPLE flavours, see link below), would it be worth raising an issue to pick this up later or OK to leave as-is?

Also, I see that roughpy has different module name convention. Would you prefer this was called `RoughPyJax_PyModule` rather than `RoughPyJaxModule`?

https://github.com/datasig-ac-uk/RoughPy/blob/0eb036bf74042cb0bf3421d2ee5a14996a632bbd/roughpy/CMakeLists.txt#L140